includes/formFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -604,12 +604,16 @@ private static function buildDescriptorNamespaces(
 
 		foreach ( $nsID as $name => $id ) {
 			$namespaceData = $mwNamespace->list( $id );
+			
+			$request = new WebRequest;
```

Comment:
This is always a blank web request

Reply:
Do we also need to check this if the namespace exists? Seems like it might modify things unexpectedly

Reply:
the webrequest class is just loaded in order to use it to get the value of the post without $_POST as that is discouraged in mediawiki. As in the second part I am not completely sure what you mean?

Reply:
Getting the WebRequest object from context would make more sense.

For the second point, what if WebRequest isn’t defined? Where would the information come from?